src/swap/do_action.rs

```diff
@@ -16,14 +21,27 @@ where
             return Ok(Next::Continue(event));
         }
 
-        if self.should_abort().await? {
-            return Ok(Next::Abort);
-        }
+        let should_abort = async {
+            loop {
+                if self.should_abort().await.unwrap_or(true) {
```

Comment:
From the perspective of this function `ShouldAbort` is just a future that resolves into a `Result` of a `bool`. 

> The issue here is that part of the should_abort logic is in the implementation of ShouldAbort (get blockchain time and compare with expiry). Another part is here (default to true).

I don't think handling this error is part of the logic of `ShouldAbort`. Handling the error in this case does result in aborting, but we could just bubble up the error or retry.

> Either put all the logic in the implementation of ShouldAbort or rename ShouldAbort to what it really is: ExpiryTimePassed.

I will rename it to `BetaHasExpired` for simplicity and because we're in Nectar.